pkg/controller/postgres/postgres_controller.go

```diff
@@ -272,6 +274,119 @@ func (r *ReconcilePostgres) ensureServicesExist(postgres *contrail.Postgres) (le
 	return leaderIP, replicaSvc.EnsureExists()
 }
 
+func (r *ReconcilePostgres) ensureServiceAccountAndRoleExist(accountName string, namespace string, postgres *contrail.Postgres) error {
+	serviceAccountName := "serviceaccount-" + accountName
+	clusterRoleName := "clusterrole-" + accountName
+	clusterRoleBindingName := "clusterrolebinding-" + accountName
+
+	serviceAccount := &core.ServiceAccount{
+		TypeMeta: meta.TypeMeta{
+			APIVersion: "v1",
+			Kind:       "ServiceAccount",
+		},
+		ObjectMeta: meta.ObjectMeta{
+			Name:      serviceAccountName,
+			Namespace: namespace,
+		},
+	}
+
+	if _, err := controllerutil.CreateOrUpdate(context.Background(), r.client, serviceAccount, func() error {
+		return controllerutil.SetControllerReference(postgres, serviceAccount, r.scheme)
+	}); err != nil {
+		return err
+	}
+
+	existingClusterRole := &rbac.ClusterRole{}
+	err := r.client.Get(context.TODO(), types.NamespacedName{Name: clusterRoleName}, existingClusterRole)
+	if errors.IsNotFound(err) {
+		clusterRole := &rbac.ClusterRole{
+			TypeMeta: meta.TypeMeta{
+				APIVersion: "rbac/v1",
+				Kind:       "ClusterRole",
+			},
+			ObjectMeta: meta.ObjectMeta{
+				Name: clusterRoleName,
+			},
+			Rules: []rbac.PolicyRule{
+				{
+					APIGroups: []string{""},
+					Verbs: []string{
+						"create",
+						"get",
+						"list",
+						"patch",
+						"update",
+						"watch",
+						"delete",
+						"deletecollection",
+					},
+					Resources: []string{"configmaps"},
+				},
+				{
+					APIGroups: []string{""},
+					Verbs: []string{
+						"get",
+						"patch",
+						"update",
+						"create",
+						"list",
+						"watch",
+						"delete",
+						"deletecollection",
+					},
+					Resources: []string{"endpoints"},
+				},
+				{
+					APIGroups: []string{""},
+					Verbs: []string{
+						"get",
+						"list",
+						"patch",
+						"update",
+						"watch",
+					},
+					Resources: []string{"pods"},
+				},
+			},
+		}
+		if err = r.client.Create(context.TODO(), clusterRole); err != nil {
+			return err
+		}
+	}
+	if err != nil {
+		return err
+	}
+
+	existingClusterRoleBinding := &rbac.ClusterRoleBinding{}
+	err = r.client.Get(context.TODO(), types.NamespacedName{Name: clusterRoleBindingName}, existingClusterRoleBinding)
+	if errors.IsNotFound(err) {
+		clusterRoleBinding := &rbac.ClusterRoleBinding{
+			TypeMeta: meta.TypeMeta{
+				APIVersion: "rbac/v1",
+				Kind:       "ClusterRoleBinding",
+			},
+			ObjectMeta: meta.ObjectMeta{
+				Name: clusterRoleBindingName,
+			},
```

Comment:
Shouldn't there be a namespace here? Because now the permissions are granted for objects across all namespaces. Is that intentional?

Reply:
No, no, clusterrole is not namespace-scoped, the subject is namespace-scoped

Reply:
Hmm, I found this example in k8s docs and it looks like the binding has to be namespace-scoped for the permissions to be granted only in that namespace. Have a look. But perhaps I'm wrong.

https://kubernetes.io/docs/reference/access-authn-authz/rbac/#rolebinding-example
```
apiVersion: rbac.authorization.k8s.io/v1
# This role binding allows "dave" to read secrets in the "development" namespace.
# You need to already have a ClusterRole named "secret-reader".
kind: RoleBinding
metadata:
  name: read-secrets
  #
  # The namespace of the RoleBinding determines where the permissions are granted.
  # This only grants permissions within the "development" namespace.
  namespace: development
subjects:
- kind: User
  name: dave # Name is case sensitive
  apiGroup: rbac.authorization.k8s.io
roleRef:
  kind: ClusterRole
  name: secret-reader
  apiGroup: rbac.authorization.k8s.io
```

Reply:
but it is a rolebinding what you send and  it is namespace-scoped, I create CLUSTERROLEBINDING and it is not namespace-scoped

Reply:
Right, my bad, sorry :D